File: semiordered_set.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] numbers = new int[n];
        int counter = 0;
        for(int i = 0 ; i < n ; i++){
            if(scanner.nextInt() != i+1) counter++;
        }
        if(counter == 2) System.out.println("YES");
        else System.out.println("NO");
    }
}

File: semiordered_set.cpp
#include<iostream>
#include<vector>
using namespace std;

int main()
{
  int num;
  cin >> num;
  vector<int> numbers(num);
  int check=0;
  numbers[num]=99999999;
  for(int i=0;i<num;i++)
  {
    cin>>numbers[i];
  }
  for(int i=0;i<num;i++)
  {
    if(numbers[i] > numbers[i+1])
    {
      check++;
    }
  }
  if(check==2)
  {
    cout << "YES";
  }
  else{
    cout << "NO";
  }
}
